Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,434 @@
+
+        #region DependencyGraphTrees
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphOverClocked =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(OverClocked),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(OverClocker),
+                        new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentPerson))
+                        )
+                );
+
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphAttack =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(TotalAttack),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(Attack),
+                        new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentPerson))
+                        ),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(AttackMod),
+                        _graphOverClocked
+                    )
+                );
+
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphSleaze =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(TotalSleaze),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(Sleaze),
+                        new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentPerson))
+                        ),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(SleazeMod),
+                        _graphOverClocked
+                    )
+                );
+
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphDataProcessing =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(TotalDataProcessing),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(DataProcessing),
+                        new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentPerson))
+                        ),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(DataProcessingMod),
+                        _graphOverClocked
+                    )
+                );
+
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphFirewall =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(TotalFirewall),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(Firewall),
+                        new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentPerson))
+                        ),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(FirewallMod),
+                        _graphOverClocked
+                    )
+                );
+
+        private static readonly DependencyGraphNode<string, MatrixLogic> _graphCurrentAction =
+                new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentAction),
+                    new DependencyGraphNode<string, MatrixLogic>(nameof(CurrentActionIndex))
+                );
```

Comment:
So the nice thing about Chummer's dependency graphs is that you don't need to do all this. So long as you register once in some place that one property A depends on another B, every single place where you register dependence on property A will also automatically register dependence on property B.